tests/deploy/test_scheduler_agent.py

```diff
@@ -1232,6 +1215,67 @@ def make_resources(
     assert agent.executor_manager.executors["agent2"].execute_count == 3
     assert agent.executor_manager.executors["agent3"].execute_count == 0
 
+    #########################
+    # Verify failure events #
+    #########################
```

Comment:
I see why you moved this, but honestly I don't think you can contract these two scenarios.

The existing one verifies that we send failure events when event propagation is *enabled*.

The new one verifies that we send recovery events *even if event propagation is disabled*. I'd expect to start this one with the exact opposite assertion as for the first test, i.e. set up r1 to deploy with events disabled and assert that r2 does *not* get deployed. Then deploy it, recover r1, and verify that now r2 does get deployed for the recovery.

Reply:
I see that at the bottom of this file you've got a similar scenario. If you make sure that event propagation is disabled there, that could suffice. Do you see a reason to have a scenario in this test case as well?